retrieval/aggregator.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retrieval
+
+import (
+	"context"
+	"math"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	promlabels "github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/tsdb/labels"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+)
+
+var statsRecord = stats.Record
+
+const metricPrefix = "prometheus_sidecar/aggregated_counters/"
```

Comment:
I have concern on adding additional prefix. As name of aggreated counter metrics are defined by user, I would make metric name simple and do not add prefix.

Reply:
Given that the sidecar also exports some internal metrics with the `prometheus_sidecar/` prefix, I think adding another path element will make the origin of the metrics slightly more obvious, and also name collisions less likely. I don't feel strongly about this, so happy to change.

Are you suggesting to remove the `aggregated_counters` part and use the `prometheus_sidecar/` prefix, or no prefix at all?

Reply:
The `prometheus_sidecar` are metrics reported by the sidecar itself.

I agree with @StevenYCChou, we shouldn't modify the metric name here. If the user wants a prefix, they can include it in the metric name in the configuration.

Reply:
Removed the prefix.